src/bio/bio_device.c

```diff
@@ -777,13 +779,17 @@ led_device_action(void *ctx, struct spdk_pci_device *pci_device)
 }
 
 static int
-check_faulty_devs(struct bio_xs_context *xs_ctxt, bool *is_faulty, struct spdk_pci_addr pci_addr)
+set_timer_and_check_faulty(struct bio_xs_context *xs_ctxt, struct spdk_pci_addr pci_addr,
```

Comment:
Seems like a case of a function being expanded to do two totally different things. Based on how it's called, it looks like we do either one or the other, not both. It's good to share what can be shared, but I wonder if there is another way to refactor using helper functions to share instead.

Reply:
This helper function wraps a workflow that is required for 2 activities, the set_timer and check_faulty functions use it to apply those 2 distinct activities that both use this workflow. An alternative would be to supply a function pointer to the workflow function but I think this is overkill in this case.